src/runtime/micro/micro_session.h

```diff
@@ -0,0 +1,257 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file micro_session.h
+ */
+#ifndef TVM_RUNTIME_MICRO_MICRO_SESSION_H_
+#define TVM_RUNTIME_MICRO_MICRO_SESSION_H_
+
+#include "micro_common.h"
+#include "micro_section_allocator.h"
+
+#include <tvm/runtime/registry.h>
+#include <tvm/runtime/c_runtime_api.h>
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "low_level_device.h"
+#include "device/utvm_runtime.h"
+#include "target_data_layout_encoder.h"
+
+namespace tvm {
+namespace runtime {
+
+/*!
+ * \brief session for facilitating micro device interaction
+ */
+class MicroSession : public ModuleNode {
+ public:
+  /*!
+   * \brief Get member function to front-end
+   * \param name The name of the function.
+   * \param sptr_to_self The pointer to the module node.
+   * \return The corresponding member function.
+   */
+  virtual PackedFunc GetFunction(const std::string& name,
+                                 const std::shared_ptr<ModuleNode>& sptr_to_self);
+
+  /*!
+   * \return The type key of the executor.
+   */
+  const char* type_key() const final {
+    return "MicroSession";
+  }
+
+  /*!
+   * \brief constructor
+   */
+  MicroSession();
+
+  /*!
+   * \brief destructor
+   */
+  ~MicroSession();
+
+  // TODO(weberlo): It'd be nice to have both `Global` and `SetGlobal` methods,
+  // but storing `curr_session` as a static class variable seems to cause
+  // undefined reference errors.  Are there alternatives?
+
+  /*!
+   * \brief get MicroSession global singleton
+   * \return pointer to the micro session global singleton
+   */
+  static std::shared_ptr<MicroSession> Global(
```

Comment:
Checkout https://github.com/dmlc/tvm/blob/master/include/tvm/build_module.h#L110

and code below https://github.com/dmlc/tvm/blob/master/src/codegen/build_module.cc#L237